blocks/procedures.js

```diff
@@ -501,6 +487,58 @@ Blockly.Blocks['procedures_mutatorarg'] = {
     field.createdVariables_ = [];
     field.onFinishEditing_('x');
   },
+  /**
+   * This will create & delete variables and in dialogs workspace to ensure
+   * that when a new block is dragged out it will have a unique parameter name.
+   * @param {!Blockly.Events.Abstract} event Change event.
+   * @this Blockly.Block
+   */
+  onchange: function(event) {
+    if (!this.workspace || this.workspace.isFlyout ||
```

Comment:
The onchange method isn't actually doing anything. Mutators don't fire create or delete events for the sub blocks in the mutator. The global workspace fires a change event instead for the block being mutated. So this part of the code will always return immediately.

Reply:
Hrmm, I'm mistaken. It looks like it does fire those events, they just aren't logged. Investigating.

Reply:
Sorry about the confusion. When I tested it it seemed like variables weren't getting deleted and since the events don't get logged I thought this wasn't running. On further testing, it's actually that if you delete more than one variable at a time only the variable for the top-most block gets removed, which causes variables to leak in the mutator workspace. For delete events you'll want to go through the list of field tags and delete all the variables that are being deleted, not just the first one, and then this should work as expected.

Testing:
1. Drag out 3 input name blocks which should create 'x', 'i', 'j' variables.
2. Delete the entire stack.
3. Drag out 3 more input name blocks.

Expected: Creates 'x', 'i', 'j' variables.

Current: Creates 'x', 'k', 'm' variables